contracts/utils/Memory.sol

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+/**
+ * @dev Helper library for deallocating memory reserved by abi.encode or low level calls.
+ */
+library Memory {
+    type FreePtr is bytes32;
```

Comment:
I think this should be just `Pointer`, it doesn't have anything to do with the free memory pointer strictly talking (e.g. the library may add functions for dealing with pointers)

```suggestion
    type Pointer is bytes32;
```